gadget/layout.go

```diff
@@ -543,6 +485,32 @@ func layOutStructureContent(gadgetRootDir string, ps *LaidOutStructure) ([]LaidO
 	return content, nil
 }
 
+// checkGadgetContentImages checks that images provided in the gadget fit the
+// declared gadget content sizes.
+func checkGadgetContentImages(gadgetRootDir string, vs *VolumeStructure) error {
+	if vs.HasFilesystem() {
+		return nil
+	}
+	if len(vs.Content) == 0 {
+		return nil
+	}
+
+	for _, c := range vs.Content {
+		fileSize, err := getImageSize(filepath.Join(gadgetRootDir, c.Image))
+		if err != nil {
+			return fmt.Errorf("structure #%d (%q): content %q: %v", vs.YamlIndex, vs.Name, c.Image, err)
+		}
+
+		if c.Size != 0 {
+			if c.Size < fileSize {
+				return fmt.Errorf("structure #%d (%q): content %q size %v is larger than declared %v", vs.YamlIndex, vs.Name, c.Image, fileSize, c.Size)
+			}
+		}
```

Comment:
Could be simplified a bit
```suggestion
		if c.Size != 0 && c.Size < fileSize {
		         return fmt.Errorf("structure #%d (%q): content %q size %v is larger than declared %v", vs.YamlIndex, vs.Name, c.Image, fileSize, c.Size)
		}
```

Reply:
Done